contracts/Oxo.sol

```diff
@@ -0,0 +1,172 @@
+pragma solidity 0.4.15;
+contract Oxo {
+
+	enum State { WaitPlayerOne, WaitPlayerTwo, WaitAccept, WinnerPlayerOne, WinnerPlayerTwo, Draw}
+    struct Game{
+		address playerOne;//1
+		address playerTwo;//2
+		uint8[9] board;
+		State state;
+		uint prizeFund;		
+	}
+	
+	uint  public gameId;
+	mapping (uint => Game) public games;
+	event Move(uint _gameId, address playerOne, address playerTwo, State state, string line0, string line1, string line2);
+	event GameOver(State state, string _result);
+	event AcceptGame(string _status);
+	event LogTransfer(address to, uint amount);
+	event BoardOutOrNotEmpty(string Error);
+	event CantMove(string _status, State state);
+	
+	modifier onlyBoardAndEmpty(uint _id, uint _field){
+		if(_field > 8){
+			BoardOutOrNotEmpty("Error: field must be <= 8");
+			return;
+		}
+		if(games[_id].board[_field] != 0){
+			BoardOutOrNotEmpty("Error: field must be empty");
+			return;			
+		}
+		_;
+	}
+
+	modifier onlyCanAccept(uint _prizeFund, uint _id){
+	    if(games[_id].state != State.WaitAccept && msg.sender != games[_id].playerTwo){
+		    return;
+		}
+		if(games[_id].prizeFund/2 > _prizeFund){
+			AcceptGame("Error: msg.value must be = prizeFund/2");
+			return;
+		}
+		_;
+	}
+	
+ 	modifier onlyWaitThisPlayer(uint _id){
+ 		if(games[_id].playerTwo == address(0)){
+			CantMove("There is no game with this id", games[_id].state);
+			return;
+		}else if(games[_id].state == State.WaitAccept && sumNotEmptyInArray(games[_id].board) != 0){
+			CantMove("WaitAccept", games[_id].state);
+			return;
+		}else if(games[_id].state == State.WinnerPlayerOne || games[_id].state == State.WinnerPlayerTwo || games[_id].state == State.Draw){
+			CantMove("GameOver", games[_id].state);
+			return;
+		}else if(games[_id].state == State.WaitPlayerOne && games[_id].playerOne != msg.sender){
+			CantMove("Wait Player One", games[_id].state);
+			return;
+		}else if(games[_id].state == State.WaitPlayerTwo && games[_id].playerTwo != msg.sender){
+			CantMove("Wait Player Two", games[_id].state);
+            return;
+        }
+       _;
+	}
+	
+	function newGame(address _playerTwo, uint _field) onlyBoardAndEmpty(0, _field) public payable returns(bool) {
+	    if(msg.sender == _playerTwo || _playerTwo == address(0)){ revert();}
+		gameId = gameId + 1;
+		games[gameId].playerOne = msg.sender;
+		games[gameId].playerTwo =_playerTwo;
+		games[gameId].state = State.WaitAccept;
+		games[gameId].prizeFund = msg.value * 2;
+		move(gameId, _field);
+		return true;
+	}
+	
+	function acceptGame(uint _id, uint _field) onlyCanAccept (msg.value, _id) onlyBoardAndEmpty(_id,  _field) public  payable returns(bool){
+		if(games[_id].prizeFund / 2 < msg.value){
```

Comment:
Could you accept the game without paying? 
Here you send money back in case msg.value more than bet you. But what will happen if msg.value less than bet?

Reply:
Yes, I can accept without money if the first player did not send money into the contact when he called the function newGame();
if msg.value less than bet (1/2 prizeFund ) will happen return:


	modifier onlyCanAccept(uint _prizeFund, uint _id){
+	    if(games[_id].state != State.WaitAccept && msg.sender != games[_id].playerTwo){
+		    return;
+		}
+		if(games[_id].prizeFund/2 > _prizeFund){
+			AcceptGame("Error: msg.value must be = prizeFund/2");
+			return;
+		}
+		_;
+	}